server/supernova/CMakeLists.txt

```diff
@@ -104,31 +104,57 @@ else()
   endif()
 endif()
 
-if(APPLE OR MSYS)
+if(AUDIOAPI STREQUAL "default")
+	if(APPLE)
+		set(AUDIOAPI portaudio)
+	elseif(WIN32)
+		set(AUDIOAPI portaudio)
+	else()
+		set(AUDIOAPI jack)
+	endif(APPLE)
+endif()
+
+if(NOT AUDIOAPI MATCHES "^(jack|coreaudio|portaudio)$")
+	message(FATAL_ERROR "Unrecognised audio API: ${AUDIOAPI}")
+endif()
+
+if(AUDIOAPI STREQUAL "coreaudio")
+	set(AUDIOAPI portaudio)
+  message(STATUS "Supernova: coreaudio API not implemented, using ${AUDIOAPI} instead")
+endif()
+
+if(AUDIOAPI STREQUAL jack)
+    find_package(Jack)
+    if(NOT JACK_FOUND)
+        message(FATAL_ERROR "Jack selected as audio API, but development files not found")
+    endif()
+elseif(AUDIOAPI STREQUAL portaudio AND ((NOT WIN32) OR MSYS)) # MSYS like Apple
     find_package(Portaudio)
     if(NOT PORTAUDIO_FOUND)
-        message(FATAL_ERROR "Portaudio missing")
-    endif()
-    target_compile_definitions(libsupernova PUBLIC PORTAUDIO_BACKEND ${PORTAUDIO_DEFINITIONS})
-    target_include_directories(libsupernova PUBLIC ${PORTAUDIO_INCLUDE_DIRS})
-    target_link_libraries(libsupernova ${PORTAUDIO_LIBRARIES})
-elseif (WIN32)
-    target_compile_definitions(libsupernova PUBLIC PORTAUDIO_BACKEND -DPORTAUDIO_BACKEND)
-    target_include_directories(libsupernova PUBLIC ${PORTAUDIO_INCLUDE_DIRS})
-    if(MSYS)
-        target_link_libraries(libsupernova ${PORTAUDIO_LIBRARIES})
-    else()
-        target_link_libraries(libsupernova portaudio)
+        message(FATAL_ERROR "Portaudio selected as audio API, but development files not found")
     endif()
-else()
+endif()
+message(STATUS "Supernova audio API: ${AUDIOAPI}")
+
+if (AUDIOAPI STREQUAL jack)
```

Comment:
change this and the portaudio check below from `jack` to `"jack"` -- otherwise, cmake will first see if a variable with the name `jack` exists, which is not what you want

Reply:
this is taken from the scsynth CMakeLists:
https://github.com/supercollider/supercollider/blob/ebe77610fd745d62ba5c36780d42f5da54d35362/server/scsynth/CMakeLists.txt#L18
should this be changed in scsynth's CMakeLists as well?